sensor/kubernetes/listener/resources/selector_test.go

```diff
@@ -0,0 +1,221 @@
+			givenSelectorLabels:                 labelsEmpty,
+			matchEmptySelector:                  emptyMatchesNothing(),
+			givenMatchingLabels:                 labelsEmpty,
+			expectedMatch:                       false,
+			expectedSelectorMatchesFunctionCall: false,
+		},
+		"Empty selector with matchEmpty set to true should match everything; attempting to match some labels": {
+			givenSelectorLabels:                 labelsEmpty,
+			matchEmptySelector:                  emptyMatchesEverything(),
+			givenMatchingLabels:                 labelsFiveElements,
+			expectedMatch:                       true,
+			expectedSelectorMatchesFunctionCall: false,
+		},
+		"Empty selector with matchEmpty set to true should match everything; attempting to match empty labels": {
+			givenSelectorLabels:                 labelsEmpty,
+			matchEmptySelector:                  emptyMatchesEverything(),
+			givenMatchingLabels:                 labelsEmpty,
+			expectedMatch:                       true,
+			expectedSelectorMatchesFunctionCall: false,
+		},
+		"More selector labels than received labels -> no match and selector Matches function not called": {
+			givenSelectorLabels:                 labelsThreeElements,
+			matchEmptySelector:                  emptyMatchesEverything(),
+			givenMatchingLabels:                 labelsOneElement,
+			expectedMatch:                       false,
+			expectedSelectorMatchesFunctionCall: false,
+		},
+		"Equal number but different labels, selector Matches function should be called and return false": {
+			givenSelectorLabels:                 labelsThreeElements,
+			matchEmptySelector:                  emptyMatchesEverything(),
+			givenMatchingLabels:                 labelsThreeElements2,
+			expectedMatch:                       false,
+			expectedSelectorMatchesFunctionCall: true,
+		},
+		"Equal labels, selector Matches function should be called and return true": {
+			givenSelectorLabels:                 labelsThreeElements,
+			matchEmptySelector:                  emptyMatchesEverything(),
+			givenMatchingLabels:                 labelsThreeElements,
+			expectedMatch:                       true,
+			expectedSelectorMatchesFunctionCall: true,
+		},
+		"selector with one label, match with three labels including the one. Expected to return true and call Matches": {
+			givenSelectorLabels:                 labelsOneElement,
+			matchEmptySelector:                  emptyMatchesEverything(),
+			givenMatchingLabels:                 labelsThreeElements,
+			expectedMatch:                       true,
+			expectedSelectorMatchesFunctionCall: true,
+		},
+	}
+	for name, tt := range tests {
+		s.Run(name, func() {
+			s.hasMatchesBeenCalled = false
+			selectorWrap := createSelector(tt.givenSelectorLabels, tt.matchEmptySelector)
+			s.injectMockSelector(&selectorWrap)
+
+			receivedMatch := selectorWrap.Matches(createLabelsWithLen(tt.givenMatchingLabels))
+			receivedSelectorMatchesFunctionCalled := s.hasMatchesBeenCalled
```

Comment:
Both of them can be removed, but the first is a function call and you could find it useful when debugging. I am not sure whether this would happen, so I leave the decision up to you.

Reply:
Fair enough